### PR TITLE
[geos] Update to 3.8.1

### PR DESCRIPTION
**Pull request by oskaritimperi**
**Describe the pull request**

- What does your PR fix?

Update geos to 3.8.1

- Which triplets are supported/not supported? Have you updated the CI baseline?

No changes to triplet support I think.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

It should.

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]
Thanks for the PR! I guess this PR ready for review.

File: ports/geos/static-lib-prefix.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+index b78ca54b..26778c83 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -228,6 +228,10 @@ if(BUILD_SHARED_LIBS)
+ 
+   set_target_properties(geos PROPERTIES VERSION ${GEOS_VERSION})
+   set_target_properties(geos PROPERTIES SOVERSION ${GEOS_VERSION})
++else()
++  if(WIN32)
++    set_target_properties(geos PROPERTIES PREFIX lib)
```
Comment (ras0219-msft) [inline ports/geos/static-lib-prefix.patch:11]: Why is this needed? Ideally, the static libraries and the import libraries will have the same name (see the maintainer guide[1]). If there are downstream ports that break when only the dynamic names are provided, they should be patched instead to use the CMake targets.

[1] https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-rename-binaries-outside-the-names-given-by-upstream

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

**Comment by oswjk** [comment]
Hi, I cant take a look at this for a couple of weeks as I'm on vacation.

BR,
Oskari
________________________________

File: ports/geos/static-lib-prefix.patch (hunk 1 above)
Comment (oswjk) [reply ports/geos/static-lib-prefix.patch]: I probably just thought to make the port work like it did before. I've added a commit to fix libspatialite and gdal builds.

**Comment by oswjk** [comment]
@PhoebeHui Can I somehow run the CI again as the build errors seems to be something that I cannot control? For example [The windows build](https://dev.azure.com/vcpkg/public/_build/results?buildId=41045&view=logs&j=384c4415-93cb-5962-b64b-d976ffb39bd5&t=10350a3b-04e8-59df-137f-98051148e2b1&l=42):

```
Downloading powershell-core...
 https://github.com/PowerShell/PowerShell/releases/download/v6.2.1/PowerShell-6.2.1-win-x86.zip -> D:\downloads\PowerShell-6.2.1-win-x86.zip
WinHttpReceiveResponse() failed: 12152
##[error]Convert-Path : Cannot find path 'C:\agent\_work\1\a\xml-results\x64-windows.xml' because it does not exist.
```

**Comment by PhoebeHui** [comment]
@oswjk, the failures doesn't relate to your changes, please ignore it, I have rerun the CI testing.

File: ports/spatialite-tools/portfile.cmake [hunk 2]
```diff
@@ -33,8 +33,8 @@ if(VCPKG_CRT_LINKAGE STREQUAL dynamic)
 else()
     set(CL_FLAGS_DBG "/MTd /Zi /DACCEPT_USE_OF_DEPRECATED_PROJ_API_H")
     set(CL_FLAGS_REL "/MT /Ox /DACCEPT_USE_OF_DEPRECATED_PROJ_API_H")
-    set(GEOS_LIBS_REL "${LDIR}/lib/libgeos_c.lib ${LDIR}/lib/libgeos.lib")
-    set(GEOS_LIBS_DBG "${LDIR}/debug/lib/libgeos_c.lib ${LDIR}/debug/lib/libgeos.lib")
+    set(GEOS_LIBS_REL "${LDIR}/lib/geos_c.lib ${LDIR}/lib/geos.lib")
+    set(GEOS_LIBS_DBG "${LDIR}/debug/lib/geos_c.lib ${LDIR}/debug/lib/geos.lib")
```
Comment (ras0219) [inline ports/spatialite-tools/portfile.cmake:37]:
```suggestion
    set(GEOS_LIBS_DBG "${LDIR}/debug/lib/geos_cd.lib ${LDIR}/debug/lib/geosd.lib")
```

**Review by ras0219 — CHANGES_REQUESTED** [review]
Because this modifies gdal, libspatialite, and spatialite-tools, those ports all need to have their `Port-Version:` incremented (or introduced and set to `1` if they don't have one).

Otherwise, LGTM

**Comment by oswjk** [comment]
Merged master as `soil` was failing.

**Review by ras0219 — APPROVED** [review]

**Pull request merged**